lib/versioning/VersionID.js

```diff
@@ -0,0 +1,136 @@
+const crypto = require('crypto');
+
+// VersionID format:
+//         timestamp  sequential_position  site_id  other_information
+// where:
+// - timestamp              14 bytes        epoch in ms (good untill 5138)
+// - sequential_position    06 bytes        position in the ms slot (1B ops)
+// - site_id                05 bytes        site identifier (like PARIS)
+// - other_information      arbitrary       user input, such as a unique string
+
+// the lengths of the components in bytes
+const LENGTH_TS = 14; // timestamp: epoch in ms
+const LENGTH_SQ = 6; // position in ms slot
+const LENGTH_ST = 5; // site identifier
+
+// empty string template for the variables in a versionId
+const TEMPLATE_TS = new Array(LENGTH_TS + 1).join('0');
+const TEMPLATE_SQ = new Array(LENGTH_SQ + 1).join('0');
+const TEMPLATE_ST = new Array(LENGTH_ST + 1).join(' ');
+
+// site identifier, like PARIS, TOKYO; will be trimmed if exceeding max length
+const SITE_ID = `${process.env.SITE_ID}${TEMPLATE_ST}`.slice(0, LENGTH_ST);
```

Comment:
Could `process.env.SITE_ID` be undefined? If yes, should it be `process.env.SITE_ID || ${a_default_name}`?